node/node.go

```diff
@@ -556,6 +562,50 @@ func (n *Node) initDatabase() error {
 	if genesisHash != expectedGenesisHash {
 		return fmt.Errorf("db contains invalid genesis hash. DB Genesis: %s Generated Genesis: %s", genesisHash, expectedGenesisHash)
 	}
+
+	ok, err := n.DB.Has(sessionKey)
+	if err != nil {
+		return err
+	}
+
+	if ok {
+		sessionBytes, err := n.DB.Get(sessionKey)
+		if err != nil {
+			return err
+		}
+		session := session{}
+		if err := json.Unmarshal(sessionBytes, &session); err != nil {
+			return fmt.Errorf("failed to unmarshal session metadata: %w", err)
+		}
+
+		n.Log.Warn(
+			"detected previous ungraceful shutdown",
+			zap.Time("timestamp", session.Timestamp),
```

Comment:
I feel like logging the timestamp of the previous startup doesn't really add a ton. I'm fine leaving it, but if we do, maybe we should just encode the time as a uint64 using https://pkg.go.dev/encoding/binary#PutVarint instead of having a struct and JSON marshaling it

Reply:
The only reason I used a struct was to allow for flexibility in the future if we wanted to add anything else here